lock-keeper-tests/src/test_suites/end_to_end/test_cases/check_session.rs

```
@@ -0,0 +1,55 @@
+use colored::Colorize;
+use lock_keeper::rpc::SessionStatus;
+use lock_keeper_client::Config;
+
+use crate::{
+    config::TestFilters,
+    error::Result,
+    run_parallel,
+    test_suites::end_to_end::{operations::authenticate, test_cases::init_test_state},
+    utils::TestResult,
+};
+
+pub async fn run_tests(config: &Config, filters: &TestFilters) -> Result<Vec<TestResult>> {
+    println!("{}", "Running check session tests".cyan());
+
+    let result = run_parallel!(
+        filters,
+        check_session_with_valid_session(config.clone()),
+        check_session_with_invalid_session(config.clone())
+    )?;
+
+    Ok(result)
+}
+
+async fn check_session_with_valid_session(config: Config) -> Result<()> {
+    let state = init_test_state(&config).await?;
+    let client = authenticate(&state).await.result?;
+
+    let res = client.check_session().await;
+    assert!(matches!(
+        res,
+        Ok(SessionStatus {
+            is_session_valid: true
+        })
+    ));
+
+    Ok(())
+}
+
+async fn check_session_with_invalid_session(config: Config) -> Result<()> {
+    let state = init_test_state(&config).await?;
+    let client = authenticate(&state).await.result?;
+
+    client.logout().await.result?;
+
+    let res = client.check_session().await;
+    assert!(matches!(
+        res,
+        Ok(SessionStatus {
+            is_session_valid: false
+        })
```

Looking at this test, I'm wondering if check_status should return an error for a bad session rather than a false within the successful response? e.g. for status codes, or if we wanted to match on the whole result somewhere?

---

Never mind, based on what we talked about, the true/false works better!